test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt

```diff
@@ -115,20 +115,42 @@ class AssetsPathHandler(context: Context) : WebViewAssetLoader.PathHandler {
     }
 }
 
+@Suppress("SpellCheckingInspection")
 class MainActivity : AppCompatActivity() {
     private lateinit var host: IModelJsHost
     private var promiseName: String = ""
+    private lateinit var env: JSONObject
+
+    companion object {
+        const val BIM_CACHE_DIR = "bim_cache"
+    }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         WebView.setWebContentsDebuggingEnabled(true)
         val alwaysExtractAssets = true // for debugging, otherwise the host will only extract when app version changes
         host = IModelJsHost(this, alwaysExtractAssets, true)
         host.startup()
-        val frontend = object : MobileFrontend(host, "&standalone=true") {
+
+        var args = "&standalone=true"
+        loadEnvJson()
+        env.optString("IMJS_STANDALONE_FILENAME", "").takeIf { it.isNotEmpty() }?.let { fileName ->
+            // ensure fileName already exists in the external files
+            getExternalFilesDir(BIM_CACHE_DIR)?.let { filesDir ->
+                File(filesDir, fileName).takeIf { it.exists() }?.let {
```

Comment:
In my opinion, having two it variables so close together with completely different meanings is confusing. I would suggest using `bimFile -> bimFile.exists()` here, and `bimFilename -> bimFileName.isNotEmpty()` above.

Reply:
Oh, and `bimFile ->` at the end of the line?

Reply:
(The fact taht I'm not even sure if the `it` on the following line (141) is a File hopefully adds to my case for this being confusing.

Reply:
I refactored the code to be more readable.